### PR TITLE
Fix appending of object arrays

### LINKED ISSUE #243: bulk append of object arrays to ctable fails when slicing and repring
```
In [7]: from bcolz import ctable

In [8]: import pandas as pd

In [9]: recs = pd.DataFrame(dict(a=list('abcd'), b=np.random.randint(10, size=4))).to_records(index=False).view(np.ndarray).copy()

In [10]: c = ctable(np.array([], dtype=recs.dtype), rootdir='df')

In [11]: c.append(recs)

In [12]: c  # calls repr under the hood
ValueError: chunkfile df/a/data/__-3.blp not found

In [13]: c[:2]
ValueError: could not broadcast input array from shape (4) into shape (2)
```

### PR DESCRIPTION
**Pull request by cpcloud**
closes #243

### CODE REVIEW

**Comment by cpcloud** [comment]
@esc can i get your review on this when you get a chance?

File: bcolz/carray_ext.pyx [hunk 1]
```diff
@@ -1386,7 +1386,9 @@ cdef class carray:
 
             # Object dtype requires special storage
             if arrcpy.dtype.char == 'O':
-                self._store_obj(array)
+                n = len(arrcpy)
+                for j in range(n):
```
Comment (FrancescAlted) [inline bcolz/carray_ext.pyx]: Very minor, but you can probably just write:

```
for j in range(len(arrcpy)):
```

with no need to declare `n` and having the same speed.

**Comment by FrancescAlted** [comment]
Apart from the comment, everything looks good to me.

**Comment by FrancescAlted** [comment]
Also, please add an entry in RELEASE_NOTES.txt.

File: bcolz/carray_ext.pyx (hunk 1 above)
Comment (cpcloud) [inline bcolz/carray_ext.pyx]: cool, done

**Comment by cpcloud** [comment]
@FrancescAlted Ok, I addressed your comment and added a release note. Thanks for looking at this.

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -11,6 +11,9 @@ Changes from 0.10.0 to 0.11.0
 - Do not try to flush when an carray is opened in 'r'ead-only mode.
   (closes #241 @FrancescAlted).
 
+- Fix appending of object arrays to already existing carrays
+  (closes #243)
```
Comment (FrancescAlted) [inline RELEASE_NOTES.rst]: Come on, don't be shy and put your Github ID here (as in the other entries ;)

**Comment by cpcloud** [comment]
@FrancescAlted Okay, I added my user name :)

**Comment by FrancescAlted** [comment]
Merged.  Thanks for your contribution!

**Comment by cpcloud** [comment]
No problemo!

**Pull request merged**